influunt-api/modules/influunt-core/app/status/StatusControladorFisico.java

```diff
@@ -0,0 +1,112 @@
+package status;
+
+import models.StatusDevice;
+import org.jetbrains.annotations.NotNull;
+import org.jongo.Aggregate;
+import org.jongo.MongoCollection;
+import org.jongo.MongoCursor;
+import play.api.Play;
+import uk.co.panaxiom.playjongo.PlayJongo;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by lesiopinheiro on 9/2/16.
+ */
+public class StatusControladorFisico {
+
+    public static final String COLLECTION = "status_controladores";
+
+    public static PlayJongo jongo = Play.current().injector().instanceOf(PlayJongo.class);
+
+    public String _id;
+
+    public String idControlador;
+
+    public Long timestamp;
+
+    public StatusDevice statusDevice;
+
+    public StatusControladorFisico(String idControlador, long timestamp, StatusDevice statusDevice) {
+        this.idControlador = idControlador;
+        this.timestamp = timestamp;
+        this.statusDevice = statusDevice;
+    }
+
+    public StatusControladorFisico(Map map) {
+        this.idControlador = map.get("idControlador").toString();
+        this.timestamp = (long) map.get("timestamp");
+        this.statusDevice = StatusDevice.valueOf(map.get("statusDevice").toString());
+    }
+
+    public static MongoCollection status() {
+        return jongo.getCollection(COLLECTION);
+    }
+
+
+    public static List<StatusControladorFisico> findByIdControlador(String idControlador) {
+        return toList(status().find("{ idControlador: # }", idControlador).sort("{timestamp: -1}").as(Map.class));
+    }
+
+    public static HashMap<String, StatusDevice> ultimoStatusDosControladores() {
+        //TODO: Confirmar se o last nao pega um registro aleatorio. Ele pode ser causa de inconsitencia
+        HashMap<String, StatusDevice> hash = new HashMap<>();
+        Aggregate.ResultsIterator<Map> ultimoStatus =
+                status().aggregate("{$sort:{timestamp:-1}}").and("{$group:{_id:'$idControlador', 'timestamp': {$max:'$timestamp'},'statusDevice': {$first:'$statusDevice'}}}").
+                        as(Map.class);
+        for (Map m : ultimoStatus) {
+            hash.put(m.get("_id").toString(), StatusDevice.valueOf(m.get("statusDevice").toString()));
+        }
+
+        return hash;
+    }
+
+    public static StatusControladorFisico ultimoStatus(String idControlador) {
+        MongoCursor<Map> result = status().find("{ idControlador: # }", idControlador).sort("{timestamp:-1}").limit(1).as(Map.class);
+        if (result.hasNext()) {
+            return new StatusControladorFisico(result.next());
+        } else {
+            return null;
+        }
+    }
+
+    public static List<StatusControladorFisico> historico(String idControlador, int pagina, int quantidade) {
+        MongoCursor<Map> result = status().find("{ idControlador: # }", idControlador).sort("{timestamp:-1}").skip(pagina * quantidade).limit(quantidade).as(Map.class);
```

Comment:
Parece que o offset da query está errado. Quando for buscar a primeira página do histórico, com 10 itens por página, ele vai dar um skip nos 10 primeiros registros (`.skip(pagina * quantidade)`), sendo que não era pra skipar nenhum registro.